app/controllers/api/base_controller/generic.rb

```diff
@@ -19,6 +19,7 @@ def add_resource(type, _id, data)
             data.delete(sc.to_s)
           end
         end
+        klass = validate_type(klass, data['type']) if data['type']
```

Comment:
`descendant_get` returns the descendant, so I believe you're reassigning `klass` to the type class here, no? But if you are, how are tests passing? O_o  Confused.